Nos/Views/Components/Media/LinkPreview.swift

```diff
@@ -0,0 +1,41 @@
+import SwiftUI
+import LinkPresentation
+import SDWebImageSwiftUI
+
+/// Shows an image button or a link view, depending on the URL.
+struct LinkPreview: View {
+    /// The URL of the content to display.
+    let url: URL
+    
+    var body: some View {
+        if url.isImage {
+            ImageButton(url: url)
+        } else {
+            LPLinkViewRepresentable(url: url)
+        }
+    }
+}
+
+/// A view that displays an Open Graph Protocol preview of the given URL.
+struct LPLinkViewRepresentable: UIViewRepresentable {
```

Comment:
Given that it is used in two different Views, we could move this struct to a different file to keep things more organized.

Reply:
Same as above -- I didn't add this or change anything here. I'll move it back where it was and see how the diff looks.